### ISSUE TITLE
Is it possible to manually kill the process for a command?

### ISSUE THREAD

**Issue by jakst** [steveukx/git-js#746]
I would like to be able to kill the process of some commands early. Is that possible?

Currently, if five commands are launched in tight sequence I dedupe the response handler and throw away the first four. But the underlying git processes still go to completion, eating up CPU in the meantime and slowing down getting the result from the fifth invocation. So basically I want to kill the previous child process when a new one launches (but with manual controls, because I still want different commands to run in parallel).

**Comment by steveukx** [comment]
Hi, apologies it's taken a while to get a solution for this.

From the next version of `simple-git` you will be able to use an `AbortController` in the options passed into the `simpleGit` instance.

https://github.com/steveukx/git-js/blob/main/docs/PLUGIN-ABORT-CONTROLLER.md

**Comment by jakst** [comment]
No worries, thanks for implementing it! 👌😊

**Comment by shumadrid** [comment]
Hi, from the example it seems that it's not possible to abort individual commands?

**Comment by steveukx** [comment]
Have you tried the abort plugin? If you have and are having problems, please open a new issue, including the code you've tried error you're seeing.

**Comment by shumadrid** [comment]
Hi, thanks for the quick response.

No, I didn't dive deep into the code to test this, that's why i didn't open a new issue.

I'm wondering if it's possible to abort individual commands instead of calling abort on the whole simplegit instance. (From first glance, it doesn't seem to be possible)
